gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -60,36 +58,94 @@
   protected static final String TASK_STATE_STORE_TABLE_SUFFIX = ".tst";
   protected static final String JOB_STATE_STORE_TABLE_SUFFIX = ".jst";
 
-  // Framework configuration properties
+  // System configuration properties
   protected final Properties properties;
 
-  // Store for persisting job state
+  // State store for persisting job states
   private final StateStore<JobState> jobStateStore;
 
-  // Job history store that stores job execution information
+  // Store for runtime job execution information
   private final Optional<JobHistoryStore> jobHistoryStore;
 
-  public AbstractJobLauncher(Properties properties)
+  // Job configuration properties
+  protected final Properties jobProps;
+  protected final String jobName;
+  protected final String jobId;
+  protected final JobState jobState;
+  private final JobCommitPolicy jobCommitPolicy;
+  private final boolean jobLockEnabled;
+  private final Optional<JobMetrics> jobMetricsOptional;
```

Comment:
It may not be necessary, but should we consider having a separate `Job` class that contains `jobProps`, `jobName`, `jobState`, `jobCommitPolicy`, etc. There seems to be a lot of configurations related to a `Job`, so a separate `Job` class may be a better abstraction. Then `AbstractJobLauncher` has a `Job` instance variable, which makes the relationship between a `JobLauncher` and a `Job` clear. A `JobLauncher` **has a** `Job`.